biojava-core/src/main/java/org/biojava/nbio/core/search/io/Hsp.java

```diff
@@ -125,19 +129,24 @@ public boolean equals(Object obj) {
         return returnAln;
     }
     
-    private Sequence getSequence(String gappedSequenceString){
+    @SuppressWarnings("unchecked")
+    private Sequence<C> getSequence(String gappedSequenceString){
         if (gappedSequenceString == null) return null;
         
-        Sequence returnSeq = null;
+        //Sequence<C> returnSeq = null;
+        S returnSeq = null;
         String sequenceString = gappedSequenceString.replace("-", "");
         
         try {
+            // these casting are unavoidable risky operations due to the guess made on sequence string.

```

Comment:
The operation of converting a character into a particular compound set is risky. However, this is a different sort of risk; the risk that `S` won't match the actual sequence we instantiate. You can think of it as already "knowing" what S is before the method is called, so instantiating like this is taking an unavoidable risk and adding a second run-time risk.

Anyways, I wouldn't even care about this particular function if it were the only place with unchecked casting. However, it doesn't even serve its purpose of making the rest of the package type-safe.